index.js

```diff
@@ -20,39 +16,34 @@ class KafkaTransport extends Transport {
 
     this.payloads = [];
     this.numPerBatch = opts.numPerBatch;
-    this.name = opts.name;
-    this.host = opts.host;
     this.topic = opts.topic;
-    this.clientId = opts.clientId;
-    this.connected = false;
-    this.client = new KafkaClient({ kafkaHost: this.host, clientId: this.clientId });
-    this.producer = new HighLevelProducer(this.client);
-    this.producer
-      .on("ready", () => {
-        this.connected = true;
-      })
-      .on("error", err => {
-        this.connected = false;
-        console.log(err);
-        throw new Error("Cannot connect to Kafka server");
-      });
+    this.maxPayloads = opts.maxPayloads;
+    this.producer = opts.producer;
```

Comment:
This seem to coupled to the producer. We only need the send function. What's is the intended API?

```
const options = { ??? };
winston.add(KafkaTransport, options );
```
I assume that we can only provide the send function. This is related to my previous comment. Let the user pass a send function that handles payloads and return a promise.

Reply:
We only need the send function for now, however I'd rather take the producer instead in case we need something else from the producer's API.

The winston API is something like this:

```
createLogger({
  // ...
  transports: [
    new KafkaTransport({
      topic: "/logs",
      producer: new HighLevelProducer(kafkaClient),
    })
  ]
});
```

Reply:
Ok.